### ISSUE TITLE
upload.js external trigger

### ISSUE THREAD

**Issue by jamesprices** [uikit/uikit#1779]
The Upload component is awesome. I only have one issue with its current setup. It would be handy to be able to trigger the upload from an external event in addition to the current functionality. Eg. a submit button. My current workaround is not to bind the uploadSelect until the user hits the submit button and then trigger the change event.

**Comment by arysom** [comment]
How do you do that? Much interested :)

**Comment by jamesprices** [comment]
Here's what I did:

`function doUpload(){
	var progressbar = $("#progressbar"),
    bar         = progressbar.find('.uk-progress-bar'),
    settings    = {

	    action: 'upload.php', // upload url

	    allow : '*.(jpg|jpeg|gif|png)', // allow only images

	    type : 'json',

	    params : { 
	                key : 'value',
	            	otherKey : 'otherValue'
	            },

	    before: function(){
	        // Do something before upload
	    },

	    loadstart: function() {
	        bar.css("width", "0%").text("0%");
	        progressbar.removeClass("uk-hidden");
	    },

	    progress: function(percent) {
	        percent = Math.ceil(percent);
	        bar.css("width", percent+"%").text(percent+"%");
	    },

	    allcomplete: function(response) {
	        bar.css("width", "100%").text("100%");

	        setTimeout(function(){
	            progressbar.addClass("uk-hidden");
	        }, 250);

	        // Do something when it's done
	    }
	};
	UIkit.Utils.xhrupload($('#upload').prop('files'), settings);
}
`

Basically replace var select = UIkit.uploadSelect($("#upload-select"), settings); with UIkit.Utils.xhrupload($('#upload-select').prop('files'), settings); and wrap it in a function so you can trigger it manually. I have a "Save" button that triggers it.

**Comment by arysom** [comment]
Thanks, that helps a lot, I also agree, that there should be an option to choose a trigger, not only on change.

**Comment by khanduras** [comment]
<noob> I had a question regarding upload.php - what would that contain? </noob>

**Comment by arysom** [comment]
Pretty much anything you can imagine:

        if(  !empty( $_FILES ) )
            {
                foreach( $_FILES[ 'files' ][ 'tmp_name' ] as $index => $tmpName )
                {
                    if( !empty( $tmpName ) && is_uploaded_file( $tmpName ) )
                    {
                        $uploaded_files[] = $_FILES['files']['name'][$index];

                        $this->file_model->insert(array(
                            'uploaded_at' => time() ,
                            'customer_email' =>  $data['email'],
                            'name'=> $_FILES['files']['name'][$index]
                        ));
                        move_uploaded_file( $tmpName, './upload/'.$_FILES['files']['name'][$index] );
                    }
                }
            }

Remember to catch file errors and sanitize file name.

**Comment by philoest** [comment]
We have released the next major version, UIkit3. To start focusing our work on the most recent release, we decided to close all the issues related to UIkit2. Please check, if these issues still persist in version 3 and if so, please open a new issue. 

Thanks for your understanding,
Your UIkit Team
